extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -254,24 +267,35 @@ export abstract class AzureAuth implements vscode.Disposable {
 		return this.getTokenHelper(tenant, resource, accessTokenString, refreshTokenString, expiresOnString);
 	}
 
-	public async getTokenHelper(tenant: Tenant, resource: Resource, accessTokenString: string, refreshTokenString: string, expiresOnString: string): Promise<OAuthTokenResponse> {
-		if (!accessTokenString) {
-			const msg = localize('azure.accessTokenEmpty', 'No access token returned from Microsoft OAuth');
-			throw new AzureAuthError(msg, 'Access token was empty', undefined);
-		}
-
-		const tokenClaims: TokenClaims = this.getTokenClaims(accessTokenString);
-		let userKey: string;
-
+	public getUserKey(tokenClaims: TokenClaims): string {
 		// Personal accounts don't have an oid when logging into the `common` tenant, but when logging into their home tenant they end up having an oid.
 		// This makes the key for the same account be different.
 		// We need to special case personal accounts.
+
+		let userKey: string;
 		if (tokenClaims.idp === 'live.com') { // Personal account
 			userKey = tokenClaims.unique_name ?? tokenClaims.email ?? tokenClaims.sub;
 		} else {
 			userKey = tokenClaims.home_oid ?? tokenClaims.oid ?? tokenClaims.unique_name ?? tokenClaims.email ?? tokenClaims.sub;
 		}
 
+		if (!userKey) {
+			Logger.pii(tokenClaims);
+			throw new AzureAuthError(localize('azure.userKeyUndefined', "User key was undefined - could not create a userKey from the tokenClaims"), 'user key undefined', undefined);
+		}
+
+		return userKey;
+	}
+
+	public async getTokenHelper(tenant: Tenant, resource: Resource, accessTokenString: string, refreshTokenString: string, expiresOnString: string): Promise<OAuthTokenResponse> {
+		if (!accessTokenString) {
+			const msg = localize('azure.accessTokenEmpty', 'No access token returned from Microsoft OAuth');
+			throw new AzureAuthError(msg, 'Access token was empty', undefined);
+		}
+
+		const tokenClaims: TokenClaims = this.getTokenClaims(accessTokenString);
+		const userKey = this.getUserKey(tokenClaims);
+
 		if (!userKey) {
```

Comment:
if userKey is undefined then this.getUserKey always throws. So this check seems unnecessary.